src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -84,7 +81,8 @@ protected override async Task LoadComponentsAsync(CancellationToken cancellation
         {
             var workspace = ComponentModel.GetService<VisualStudioWorkspace>();
 
-            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+            // Do the MEF loads and initialization in the BG explicitly.
+            await TaskScheduler.Default;
```

Comment:
It's odd to see this after the `ComponentModel.GetService<VIsualStudioWorkspace>` line, since that was already doing MEF loads. Either move to the top or just delete, if we were already on the thread pool thread to start.

Reply:
doine.